checks/evaluation/maintained.go

```diff
@@ -30,69 +33,72 @@ const (
 )
 
 // Maintained applies the score policy for the Maintained check.
-func Maintained(name string, dl checker.DetailLogger, r *checker.MaintainedData) checker.CheckResult {
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
-		return checker.CreateRuntimeErrorResult(name, e)
+func Maintained(name string,
+	findings []finding.Finding, dl checker.DetailLogger,
+) checker.CheckResult {
+	// We have 7 unique probes, each should have a finding.
+	expectedProbes := []string{
+		archived.Probe,
+		activityOnIssuesByCollaboratorsMembersOrOwnersInLast90Days.Probe,
+		commitsInLast90Days.Probe,
+		wasCreatedInLast90Days.Probe,
 	}
 
-	if r.ArchivedStatus.Status {
-		return checker.CreateMinScoreResult(name, "repo is marked as archived")
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
+		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
-	// If not explicitly marked archived, look for activity in past `lookBackDays`.
-	threshold := time.Now().AddDate(0 /*years*/, 0 /*months*/, -1*lookBackDays /*days*/)
-	commitsWithinThreshold := 0
-	for i := range r.DefaultBranchCommits {
-		if r.DefaultBranchCommits[i].CommittedDate.After(threshold) {
-			commitsWithinThreshold++
-		}
+	if projectIsArchived(findings) {
+		checker.LogFindings(nonNegativeFindings(findings), dl)
+		return checker.CreateMinScoreResult(name, "project is archived")
 	}
 
-	// Emit a warning if this repo was created recently
-	recencyThreshold := time.Now().AddDate(0 /*years*/, 0 /*months*/, -1*lookBackDays /*days*/)
-	if r.CreatedAt.After(recencyThreshold) {
-		dl.Warn(&checker.LogMessage{
-			Text: fmt.Sprintf("repo was created in the last %d days (Created at: %s), please review its contents carefully",
-				lookBackDays, r.CreatedAt.Format(time.RFC3339)),
-		})
-		daysSinceRepoCreated := int(time.Since(r.CreatedAt).Hours() / 24)
-		return checker.CreateMinScoreResult(name,
-			fmt.Sprintf("repo was created %d days ago, not enough maintenance history", daysSinceRepoCreated),
-		)
+	if projectWasCreatedInLast90Days(findings) {
+		checker.LogFindings(nonNegativeFindings(findings), dl)
+		return checker.CreateMinScoreResult(name, "project was created in last 90 days. please review its contents carefully")
 	}
 
+	commitsWithinThreshold := 0
 	issuesUpdatedWithinThreshold := 0
-	for i := range r.Issues {
-		if hasActivityByCollaboratorOrHigher(&r.Issues[i], threshold) {
-			issuesUpdatedWithinThreshold++
+
+	for i := range findings {
+		f := &findings[i]
+		if f.Outcome == finding.OutcomePositive {
+			switch f.Probe {
+			case activityOnIssuesByCollaboratorsMembersOrOwnersInLast90Days.Probe:
+				issuesUpdatedWithinThreshold++
+			case commitsInLast90Days.Probe:
+				commitsWithinThreshold++
+			}
```

Comment:
i personally think this is unwieldy. I'd prefer a numerical probe. What do you think @laurentsimon

Reply:
@laurentsimon kind ping.

Reply:
I commented in https://github.com/ossf/scorecard/pull/3507#discussion_r1355363786 wdut?

Reply:
Renamed in https://github.com/ossf/scorecard/pull/3507/commits/d7938f44a1736f9ffb47506ccb021922ef118303